### ISSUE TITLE
Chunks removeing all AE things

### ISSUE THREAD

**Issue by andr9528** [thiakil/Applied-Energistics-2#76]
Minecraft Version: 1.12.1
AE Version: Applied Llamagistics 1.12-rv5-beta-7

I have now experienced 3 times that parts of my system simply vanishes, 2 times it has hit my importer on my actually additions farmer, the remaining time it just hit some non-essential wires (had a duplicate route for the system through another chunk)

![2017-09-17_11 28 09](https://user-images.githubusercontent.com/21788044/30519635-92f57a6e-9b9b-11e7-80cb-3c393df9d276.png)
![2017-09-17_11 28 28](https://user-images.githubusercontent.com/21788044/30519636-92f5b538-9b9b-11e7-81fb-cc8a24e913b3.png)

As seen on the pictures it precisely 1 chunk where everything AE vanishes.
I'm living in a Void World from the Simple Void World mod, but if it was that mod that was causing a issue, i would have thought it to not be isolated to AE blocks.

If it is related to using Applied Llamagistics, or you are not the developer for that mod, then tell me so, as their curse direct to this issue site.

**Comment by thiakil** [comment]
This is an issue with (probably) some other mod, and no one has (to my knowledge) worked out what it is.

It causes NBT loss, which means AE/AL multipart blocks to not be saved, TE/TF blocks to reset to basic variants, etc.

**Comment by thiakil** [comment]
Logs that contain any errors during chunk saving/loading would help, but so far no one has spotted anything

**Comment by jnouri** [comment]
I've had this happen as well - I've found that if the dimension is not constantly chunk loaded it will get hit by this (among other things related to NBT data) when you load up the dimension again. 

This happens in all the 1.12 modpacks I've played.

**Comment by SingularTier** [comment]
See the referenced post for more information.

Removing FTBUtils and FTBLib and loading from a backup fixed this for our world.

Edit: Nevermind. I reproduced the issue with FTBUtils and FTBlib removed from the pack. I'll keep investigating.

**Comment by andr9528** [comment]
Could you describe how you reproduced it? @SingularTier  It might help locate where the problem comes from

**Comment by thiakil** [comment]
Yeah that would really help. So would *ANYONE* experiencing this listing their installed mods (preferably by modid in a gist/pastebin), so we can find the overlapping ones.

**Comment by andr9528** [comment]
Minecraft Version: 1.12.1
Forge Version: forge-14.22.0.2474

i took pictures of the whole modlist from the twitch program and added the pictures to this album.

https://imgur.com/a/bX1bL

hope it helps

**Comment by thiakil** [comment]
I can only see one image there mate, text format is preferable. If you trigger a forced crash (check the MC wiki for the keybind) that shoujld have a nice text list of loaded mods

**Comment by andr9528** [comment]
Unsure why you only see one image (i see all 7)...
But as requested here is the modlist in text format (whole crash report)

https://pastebin.com/ACDYFE9r

For future reference you force crash the game with F3 + C, holding it down for a few seconds

**Comment by thiakil** [comment]
When I saw it it was a single embedded image lol.

If you have a reproducible situation, I would recommend removing all other mods, trying to reproduce by adding mods one by one until it happens.

**Comment by SingularTier** [comment]
It's reproducible only in mature world we've been playing. I've been doing @thiakil 's suggestion but it's a slow process.

Here's the modlist for the world: https://pastebin.com/PdnZqwT2

To reproduce I just load in to the game after a server restart. The second I load I hear all the ME cables snapping around me.

**Comment by thiakil** [comment]
@SingularTier the damage may already be done, you'll need to check it with NBTExplorer though. If the NBT was never saved, it will always break on load (because it's like "oh bother, I have no parts/cables attached").

**Comment by SingularTier** [comment]
I find that if I move my character far enough away from the chunk and copy my position data back in to the backup, then load the backup with my character away from the ME chunk (i.e. start the world without the AE stuff chunk loaded) the NBT data is preserved. If I go to the chunk later everything is fine (as long as I don't leave the dimension and come back).

**Comment by andr9528** [comment]
Comparing our 2 lists, should allow for some mods to be ignored while looking for the cause of the issue, as from what i have seen, we do have some mods that are unique to one another.

**Comment by SingularTier** [comment]
I'm afraid it's something more sinister than a broken mod interaction, I was able to reproduce it twice [with this mod list](https://pastebin.com/X4Tetcx2) after just adding back in forestry and binnie's mods. **But I am unable to reproduce it reliably with that same mod list. I would not go pointing fingers at forestry or binnie's.** 

Actually, what makes it reproducible 100% is adding a TON of entities in the player loading chunk, and then logging out and back in to the server.  Sounds like a weird timing issue maybe? I don't know enough about mod development to be sure though. I'll try and narrow things down more. 

If anyone knows a way to slow down the server through JVM trickery I think that would help reproduction.

Edit: https://youtu.be/cbab_op9r9o Video of me reproducing it with the mod list and method mentioned.

**Comment by SingularTier** [comment]
I was able to reproduce it using only forge and Applied llamagistics: https://pastebin.com/LX5asviC

Notice all the chickens? 
Reproduction: https://youtu.be/Cw-BaNOmeFc

MY GUESS as to how to reproduce locally: 
Create a decently large AE network (in my case I have several hundred channels going)
Slow down the server login somehow (in my case, chickens)
Log in from a freshly booted client (reconnecting will not trigger bug most of the time).

**Comment by andr9528** [comment]
Good to see it being reproduced, but my case happen to be quite different from yours...

- Got a relatively small AE system (maybe 34 channels MAX, if i where using channels...)
- Mine is in SSP, while yours is in SMP, might be irrelevant but it is a difference
- Mine occurred just by swapping dimension (i live a void world from the "Simple Void World" mod) not by logging out and in, and also 2 times in the same client instance (with some time in between)
- My dimension ran at optimal speed, with no other entities in it other than me (when i'm there) and also very few chunk updates (only have one 9*9 farm automated with Actually Additions' Farmer)
- My platform in the dimension is constantly chunk loaded by FTBUtil

**Comment by SingularTier** [comment]
I also experience it swapping dimensions, but I found that if I didn't chunk load the dimension it happened MORE often.

I also had (non-AE) NBT data wiped in Modern Skyblock 2 as well, which is more congruent with what you're seeing as it's also a void world w/ minimal entities.

My guess is that it has nothing to do with the entities or AE network size, but rather some timing issue with loading chunks. Slower servers/computers or those with restricted amounts of memory might hit it more often? The entities and larger networks just slow it down enough to trigger the bug.

Oh, and I just got it to wipe TE machine NBT data without Applied Llamagistics installed. Crash report: https://pastebin.com/9rKSe7PP

This is smelling like a forge bug.

**Comment by thiakil** [comment]
Interesting. Nice work on narrowing it down!

**Comment by tyler489** [comment]
Yes this happens randomly with MEkanism machines. Thermal Expansion machines and Botania Mana pools all losing blockstates

**Comment by thiakil** [comment]
This is not specific to any one mod, Please see the linked forge issue above.
